File: triatlon-cs/Model/Activity/AthletePoints.cs
namespace triatlon_cs.Model.Activity;

// AthletePoints

public class AthletePoints : Entity<int>
{
    
    // Properties

    private int _athleteId;
    private int _points;

    public int AthleteId
    {
        get => _athleteId;
        set => _athleteId = value;
    }
    public int Points
    {
        get => _points;
        set => _points = value;
    }
    
    // Lifecycle

    public AthletePoints(int athleteId)
    {
        _athleteId = athleteId;
        _points = 0;
    }

    public AthletePoints(int athleteId, int points)
    {
        _athleteId = athleteId;
        _points = points;
    }

}

File: triatlon-java/src/main/java/model/activity/AthletePoints.java
package model.activity;

import model.Entity;

// AthletePoints

public class AthletePoints extends Entity<Integer> {

    // Private Properties

    private Integer athleteId;
    private Integer points;

    // Lifecycle

    public AthletePoints(Integer athleteId) {
        this.athleteId = athleteId;
        this.points = 0;
    }

    public AthletePoints(Integer athleteId, Integer points) {
        this.athleteId = athleteId;
        this.points = points;
    }

    // Getters & Setters

    public Integer getAthleteId() {
        return this.athleteId;
    }

    public Integer getPoints() {
        return this.points;
    }

    public void setAthleteId(Integer athleteId) {
        this.athleteId = athleteId;
    }

    public void setPoints(Integer points) {
        this.points = points;
    }

}
